src/nix-daemon/nix-daemon.cc

```diff
@@ -692,6 +692,10 @@ static PeerInfo getPeerInfo(int remote)
 
 #elif defined(LOCAL_PEERCRED)
 
+#if !defined(SOL_LOCAL)
+#define SOL_LOCAL 0
```

Comment:
No, that would be wrong, potentially causing the xucred structure to be filled with nonsense ([see this libvirt commit][1]). I mentioned more reasons why defining it to be 0 is right in the 7888b2ba commit message.

[1]: http://libvirt.org/git/?p=libvirt.git;a=commit;h=8079b0e0f40aa0ad19e9aec7a69143ee858321a2